src/goto-programs/builtin_functions.cpp

```diff
@@ -664,9 +626,8 @@ void goto_convertt::do_java_new_array(
   t_n->code=code_assignt(lhs, malloc_expr);
   t_n->source_location=location;
 
-  assert(ns.follow(object_type).id()==ID_struct);
   const struct_typet &struct_type=to_struct_type(ns.follow(object_type));
-  assert(struct_type.components().size()==3);
+  PRECONDITION(struct_type.components().size()==3);
```

Comment:
Consider using `is_valid_java_array` in java types if you want to validate that this is a java Array object (e.g. struct with a  base member, data member and length member).